Objects/typeobject.c

```diff
@@ -1600,33 +1653,18 @@ static PyObject *
 type_repr(PyTypeObject *type)
 {
     if (type->tp_name == NULL) {
-        // type_repr() called before the type is fully initialized
-        // by PyType_Ready().
+        // If type_repr() is called before the type is fully initialized
+        // by PyType_Ready(), just format the type memory address.
         return PyUnicode_FromFormat("<class at %p>", type);
     }
 
-    PyObject *mod, *name, *rtn;
-
-    mod = type_module(type, NULL);
-    if (mod == NULL)
-        PyErr_Clear();
-    else if (!PyUnicode_Check(mod)) {
-        Py_SETREF(mod, NULL);
-    }
-    name = type_qualname(type, NULL);
+    PyObject *name = type_fullyqualname(type, 1);
     if (name == NULL) {
-        Py_XDECREF(mod);
         return NULL;
     }
-
-    if (mod != NULL && !_PyUnicode_Equal(mod, &_Py_ID(builtins)))
-        rtn = PyUnicode_FromFormat("<class '%U.%U'>", mod, name);
-    else
-        rtn = PyUnicode_FromFormat("<class '%s'>", type->tp_name);
-
-    Py_XDECREF(mod);
+    PyObject *result = PyUnicode_FromFormat("<class '%U'>", name);
```

Comment:
Same comment as above: you are readding "builtins." to builtin types.

Reply:
The fully qualified name omits the module is the module is "builtins". There is no behavior change. This code has multiple tests. I'm not sure that I get your comment correctly.

Reply:
As you can see from the code, the "builtins." part was deliberately omitted for builtin types.

Reply:
I'm not sure of what you are talking about. You are commenting `type_repr()` function. I didn't change the function output, it's the same:

```
>>> import builtins
>>> print(repr(builtins.str))  # <=== HERE "builtins." is omitted
<class 'str'>
>>> builtins.str.__module__
'builtins'
>>> builtins.str.__qualname__
'str'
```

`repr()` omits the module if it's "builtins". But it's added otherwise:

```
>>> import collections
>>> print(repr(collections.OrderedDict))
<class 'collections.OrderedDict'>
>>> collections.OrderedDict.__module__
'collections'
>>> collections.OrderedDict.__qualname__
'OrderedDict'
```

It's the same behavior than before.